sycl/source/detail/queue_impl.cpp

```diff
@@ -339,10 +341,51 @@ event queue_impl::submit_impl(const detail::type_erased_cgfo_ty &CGF,
 
   HandlerImpl->MEventMode = SubmitInfo.EventMode();
 
-  auto Event = finalizeHandler(Handler, SubmitInfo.PostProcessorFunc());
+  auto isHostTask = Type == CGType::CodeplayHostTask;
+
+  // TODO: we shouldn't need to treat ExecCommandBuffer differently,
+  // but currently UR (legacy L0 adapter) does not respect
+  // in-order property of command buffer and still requires sync points.
+  auto isGraphSubmission = Type == CGType::ExecCommandBuffer;
```

Comment:
I'm not sure I understand this comment. The in-order property of a graph or command-buffer only affects the relationship between commands in that command-buffer, not how the command-buffer as a whole executes with respect to other commands in a given UR queue. "Sync points" is also generally used to refer to synchronization between commands in a command-buffer.

I think what is trying to be conveyed here is that the path in the V1 adapter where immediate append isn't available still requires event dependencies even in an in-order queue, is that right?